micro_manager/model_adaptivity/model_adaptivity.py

```diff
@@ -0,0 +1,350 @@
+"""
+Class ModelAdaptivity provides methods to change micro simulation resolution on the fly.
+"""
+from typing import Union, Optional
+
+from ..config import Config
+from ..micro_simulation import create_simulation_class
+from micro_manager.tools.logging_wrapper import Logger
+
+import numpy as np
+import importlib
+
+
+class ModelAdaptivity:
+    def __init__(self, configurator: Config, rank: int) -> None:
+        """
+        Class constructor.
+
+        Parameters
+        ----------
+        configurator : object of class Config
+            Object which has getter functions to get parameters defined in the configuration file.
+        rank : int
+            Rank of the MPI communicator.
+        """
+        self._logger = Logger("ModelAdaptivity", "./ModelAdaptivity.log", rank)
+        self._resolution_logger = Logger(
+            "ModelAdaptivity-Res", "./ModelAdaptivity-resolution.log", rank
+        )
+
+        self._model_files = configurator.get_model_adaptivity_file_names()
+        self._switching_fun_name = (
```

Comment:
```suggestion
        self._switching_func_name = (
```
Please also change wherever relevant below.